sublime_linter.py

```diff
@@ -329,12 +340,14 @@ def kill_active_popen_calls(bid):
 )  # type: DefaultDict[Bid, DefaultDict[LinterName, Set[FileName]]]
 
 
-def group_by_filename_and_update(window, bid, view_has_changed, linter, errors):
-    # type: (sublime.Window, Bid, ViewChangedFn, LinterName, List[LintError]) -> None
+def group_by_filename_and_update(main_view, view_has_changed, linter, errors):
+    # type: (sublime.View, ViewChangedFn, LinterName, List[LintError]) -> None
     """Group lint errors by filename and update them."""
     if view_has_changed():  # abort early
         return
 
+    bid = main_view.buffer_id()
```

Comment:
Generally, the getters `id()`, `buffer_id()`, `view.window()` are not safe, that's why we extract these and pass them down/around. (Here you removed `bid` from the function signature but you still need it and get the information from the view object. Same for `window` below, you removed it from the signature and then get a live version in a for loop.)

Reply:
A `view` is a real, live OOP thing here, all getters are actually side-effects, everything mutates. (It is somewhat save if we would run on the so-called main thread which blocks/freezes the UI, but we run basically all our code multi-threaded so the UI/the world changes underneath us.)

I think we should go for `affected_filenames_per_filename`.